### DISCUSSION TITLE
Netconf edit-config - replace

### DISCUSSION THREAD

**Discussion by bruckynetworkstest** [scrapli/scrapligo#129] [General]
Hello all, 

First, is it ok just to drop queries in here if I get stuck on something? 

I am really enjoying working with the library and progressing through converting everything I have done in Python into Go. 

I am just having issues trying to do a netconf edit-config operation=replace for any individual module for a Cisco ios-xe virtual router.

My Go script has not problems, it runs and when completed on stdout it states it has been successful. I can see on the test device me logging in, but there is no change to the config module. 

I am using a payload which is successful when i try with native NC, scrapli and nornir_scrapli/netconf.

Is it something incorrect I am applying ? I can add and delete whole modules fine and with no issues, just when trying an idempotent approach for operation replace.

```
package main

import (
	"fmt"

	"github.com/scrapli/scrapligo/driver/netconf"
	"github.com/scrapli/scrapligo/driver/options"
)

func main() {
	// create a new Netconf driver
	driver, err := netconf.NewDriver(
		"x.x.x.x",
		options.WithAuthNoStrictKey(),
		options.WithAuthUsername("xxxxx"),
		options.WithAuthPassword("xxxxx"),
		options.WithPort(830),
	)
	if err != nil {
		fmt.Printf("failed to create driver; error: %+v\n", err)
		return
	}

	// open a connection to the device
	err = driver.Open()
	if err != nil {
		fmt.Printf("failed to open driver; error: %+v\n", err)
		return
	}
	defer driver.Close()

	// configure NTP servers using Netconf edit-config
	configPayload := `
	<config xmlns:xc="urn:ietf:params:xml:ns:netconf:base:1.0">
	<native xmlns="http://cisco.com/ns/yang/Cisco-IOS-XE-native">
	<ntp>
		<server
			xmlns="http://cisco.com/ns/yang/Cisco-IOS-XE-ntp" operation="replace">
			<server-list>
				<ip-address>33.33.33.33</ip-address>
			</server-list>
			<server-list>
				<ip-address>66.66.66.66</ip-address>
			</server-list>
		</server>
	</ntp>
  </config>
  `
	_, err = driver.EditConfig("running", configPayload)
	if err != nil {
		fmt.Printf("failed to send edit-config request: %+v\n", err)
		return
	}

	fmt.Println("NTP configuration has been updated.")
}
```

**Comment by carlmontanari** [comment]
> First, is it ok just to drop queries in here if I get stuck on something?

yes of course -- and thank you for using discussions for this rather than issues, much appreciated! will try to help but no promises though :)

been a hot minute since ive thought about netconf config things. can try later but in the short term would defo recommend snagging logs and seeing if there is anything obvious in there (channel logs and/or scrapli logs). 

do you also need to commit the config maybe? dont recall just didn't see a commit so figured id ask!

**Reply by bruckynetworkstest** [reply]
Thanks for taking the time to respond and offer assistance.

I have already gone through the logs, and I did not find any clues that can help me solve the issue. Regarding the commit, I have deliberately left it out of my code because I am testing the edit-config functionality without committing the changes.

The problem I am facing is that the "replace" operation is not working as I see it working when using scrapli through Python. I have previously used the same code for the "delete" operation, and it worked well. Therefore, I would have assumed that the "replace" operation should work in a similar manner.

Anything I can do to help (if there is) please let me know.

**Reply by carlmontanari** [reply]
copy, understood! can you snag channel logs from a scrapligo vs py version of the same thing and share that? would be good to compare that. I can try to test this eventually as well but will be a bit I imagine!

**Reply by bruckynetworkstest** [reply]
Sorry, sorry, sorry, It seems to have been myself again where the issue was.

Thanks and appreciate the time and kind words.
